zanata-war/src/main/java/org/zanata/events/DocumentMilestoneEvent.java

```diff
@@ -27,29 +27,58 @@
 import lombok.Setter;
 
 import org.codehaus.jackson.annotate.JsonPropertyOrder;
-import org.zanata.common.ContentState;
 import org.zanata.common.LocaleId;
 
-import java.util.Collection;
-
 /**
+ *
+ * Event for when a document in a language reached a milestone in translations.
+ *
  * @author Alex Eng <a href="mailto:aeng@redhat.com">aeng@redhat.com</a>
  */
 @Getter
 @Setter
 @AllArgsConstructor
 @NoArgsConstructor
-@JsonPropertyOrder({ "project", "version", "docId", "locale", "milestone"})
+@JsonPropertyOrder({ "project", "version", "docId", "locale", "url", "milestone"})
 @EqualsAndHashCode
-public class DocumentMilestoneEvent extends JSONType {
+public class DocumentMilestoneEvent extends WebhookEventType {
 
     public static final String EVENT_NAME =
         "org.zanata.events.DocumentMilestoneEvent";
 
+    /**
+     * Target project slug.
+     * {@link org.zanata.model.HProject#slug}
+     */
     private String project;
+
+    /**
+     * Target project version slug.
+     * {@link org.zanata.model.HProjectIteration#slug}
+     */
     private String version;
+
+    /**
+     * Target document full path id.
+     * {@link org.zanata.model.HDocument#docId}
+     */
     private String docId;
+
+    /**
+     * Target locale id.
+     * {@link org.zanata.common.LocaleId}
+     */
     private LocaleId locale;
+
+    /**
+     * Editor url in target document.
+     * {@link org.zanata.util.UrlUtil#fullEditorDocumentUrl}
+     */
+    private String url;
```

Comment:
Could we give this a more descriptive name, like `editorDocumentUrl`?  Do we link to the editor, or to the viewer?  Can it be visited by an anonymous user, or a non-translator?